### PR TITLE
API to test inference endpoint credentials #8

### LINKED ISSUE #8: Add API to test inference endpoint credentials
Implement an API to validate that the inference endpoint credentials are working before they can be saved and used

### PR DESCRIPTION
**Pull request by amilmshaji**
Closes #8

### CODE REVIEW

File: app/loaders/ai71/loader.py [hunk 1]
```diff
@@ -36,7 +36,8 @@ def get_response(self, message) -> str:
             choice = message["choices"][0]
             if "message" in choice:
                 return choice["message"]["content"]
-
+        elif 'detail' in message:
+            return {"error" : message['detail']}
         return ""
```
Comment (agberoz) [inline app/loaders/ai71/loader.py:42]: Standardize the response of the `get_response` method

File: app/services/provider.py [hunk 2]
```diff
@@ -1,14 +1,44 @@
...
+    match inference.llm_provider:
+        case 'openai':
+            return test_inference_provider_connection(inference)
+        case 'togethor':
+            return test_inference_provider_connection(inference)
+        case 'ai71':
+            return test_inference_provider_connection(inference)
+        case _:
+            return None, "Unsupported Inference"
+    return None, "Unsupported Inference"
```
Comment (agberoz) [inline app/services/provider.py:42]: Remove the match case section and implement error handling in the model loader to restrict the use of unknown inference engines

File: app/services/connector_details.py [hunk 3]
```diff
@@ -1,10 +1,27 @@
...
 
+def test_inference_provider_connection(inference):
+    model_configs = [{
+        "unique_name": inference.name,
+        "name": inference.model,
+        "api_key": inference.apikey,
+        "endpoint": inference.endpoint,
+        "kind" : inference.llm_provider,
+    }]
+    inference_model = BaseLoader(model_configs= model_configs).load_model(inference.name)
```
Comment (agberoz) [inline app/services/connector_details.py:17]: Add a try-catch block here as the load_model may raise exception

**Review by agberoz — CHANGES_REQUESTED** [review]

File: app/services/provider.py (hunk 2 above)
Comment (amilmshaji) [reply app/services/provider.py:48]: Removed unwanted amtch section and  implement error handling in the model loader

File: app/loaders/ai71/loader.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (amilmshaji) [reply app/loaders/ai71/loader.py:42]: standardized to dict return

**Review by amilmshaji — COMMENTED** [review]
Updated the mention corrections

**Review by agberoz — APPROVED** [review]

**Pull request merged**